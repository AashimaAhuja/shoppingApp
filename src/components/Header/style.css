.app-header {
  background: #fff;
  display: flex;
  height: 50px;
  justify-content: center;
  flex-direction: column;
}
.likes-count {
  align-self: flex-end;
  margin-right: 1rem;
  border: 2px solid black;
  padding: 4px;
  border-radius: 3px;
  position: relative;
}
.liked-items {
  position: absolute;
  right: 0px;
  border: 1px solid;
  min-width: 150px;
  z-index: 1;
  top: 100%;
  margin-top: 4px;
  background: #fff;
  padding: 0.5rem;
  display: none;
}
.liked-items > dt {
  text-align: left;
  line-height: 20px;
}
.liked-items.active {
  display: block;
}
