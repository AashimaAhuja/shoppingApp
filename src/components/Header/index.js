import React, { useCallback, useState, useMemo } from "react";
import "./style.css";

const Header = ({ visibleProducts, testShowLikeditems }) => {
  const [showLikedItems, setShowLikedItems] = useState(testShowLikeditems || false);
  const onToggleLikedItems = useCallback(() => {
    setShowLikedItems((showLikedItems) => !showLikedItems);
  }, []);
  const likedProducts = useMemo(
    () => visibleProducts.filter(({ isLiked }) => isLiked),
    [visibleProducts]
  );

  return (
    <header className="app-header">
      <div className="likes-count" onClick={onToggleLikedItems}>
        <span>Likes {likedProducts.length}</span>
        <dl className={`liked-items ${showLikedItems ? " active" : ""}`}>
          {likedProducts.map(({ name, id }) => (
            <dt key={id}>{name}</dt>
          ))}
        </dl>
      </div>
    </header>
  );
};

export default Header;
