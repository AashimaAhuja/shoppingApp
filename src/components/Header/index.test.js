import { shallow } from 'enzyme';
import * as React from 'react';
import Header from './';

const defaultProps = {
  visibleProducts : [
      {
          isLiked: true,
          id: 1,
          name: 'prod 1'
      },
      {
          isLiked: true,
          id: 2,
          name: 'prod 1'
      },
      {
          isLiked: true,
          id: 3,
          name: 'prod 1'
      },
      {
          isLiked: true,
          id: 4,
          name: 'prod 1'
      },
      {
          isLiked: false,
          id: 5,
          name: 'prod 1'
      },
  ]
}

describe('<Header/> Tests', ()=>{
  it('should render Header Component', ()=>{
    const HeaderComponent = shallow(<Header visibleProducts={[]} />)
    expect(HeaderComponent).toMatchSnapshot();
  });
  it('should render Header Component with visible products', ()=>{
    const wrapper = shallow(<Header {...defaultProps} />)
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('dl').length).toEqual(1);
    expect(wrapper.find('dt').length).toEqual(4);
    expect(wrapper.find('span').length).toEqual(1);
    expect(wrapper.find('span').text()).toEqual('Likes 4');
  });
  it('should render Header Component with showLikedItems as true', ()=>{
    const wrapper = shallow(<Header {...defaultProps} testShowLikeditems={true} />)
    expect(wrapper).toMatchSnapshot(); 
    const likesCount = wrapper.find('.likes-count');
    likesCount.simulate('click');
  });
})
