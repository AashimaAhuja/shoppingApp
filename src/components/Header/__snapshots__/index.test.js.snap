// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Tests should render Header Component 1`] = `
<header
  className="app-header"
>
  <div
    className="likes-count"
    onClick={[Function]}
  >
    <span>
      Likes 
      0
    </span>
    <dl
      className="liked-items "
    />
  </div>
</header>
`;

exports[`<Header/> Tests should render Header Component with showLikedItems as true 1`] = `
<header
  className="app-header"
>
  <div
    className="likes-count"
    onClick={[Function]}
  >
    <span>
      Likes 
      4
    </span>
    <dl
      className="liked-items  active"
    >
      <dt
        key="1"
      >
        prod 1
      </dt>
      <dt
        key="2"
      >
        prod 1
      </dt>
      <dt
        key="3"
      >
        prod 1
      </dt>
      <dt
        key="4"
      >
        prod 1
      </dt>
    </dl>
  </div>
</header>
`;

exports[`<Header/> Tests should render Header Component with visible products 1`] = `
<header
  className="app-header"
>
  <div
    className="likes-count"
    onClick={[Function]}
  >
    <span>
      Likes 
      4
    </span>
    <dl
      className="liked-items "
    >
      <dt
        key="1"
      >
        prod 1
      </dt>
      <dt
        key="2"
      >
        prod 1
      </dt>
      <dt
        key="3"
      >
        prod 1
      </dt>
      <dt
        key="4"
      >
        prod 1
      </dt>
    </dl>
  </div>
</header>
`;
